tests/unit/iteratorFilter.cpp

```diff
@@ -0,0 +1,385 @@
+#include <graphblas/utils/iterators/filter.hpp>
+
+
+template< typename FwdIt >
+void testOneOut(
+	const size_t &n, const size_t filtered, const size_t expected,
+	FwdIt &it, const FwdIt &end,
+	grb::RC &rc
+) {
+	rc = grb::SUCCESS;
+	size_t count = 0;
+	for( ; it != end; ++it, ++count ) {
+		if( *it != *(it.operator->()) ) {
+			std::cerr << "Error: dereference operator returns " << *it << ", "
+				<< "but dereferencing the pointer operator returns " << (*(it.operator->()))
+				<< "\n";
+			rc = grb::FAILED;
+			return;
+		}
+		if( *it == filtered ) {
+			std::cerr << "Error: found " << filtered << ", "
+				<< "which should have been filtered out\n";
+			rc = grb::FAILED;
+			return;
+		}
+	}
+	if( count != expected ) {
+		std::cerr << "Error: count should be " << n << ", "
+			<< "but the filtered iterator returned " << count << " elements instead\n";
+		rc = grb::FAILED;
+	}
+}
+
+template< typename FwdIt >
+void testOnlyOne(
+	const size_t &n, const size_t retained, const size_t expected,
+	FwdIt &it, const FwdIt &end,
+	grb::RC &rc
+) {
+	rc = grb::SUCCESS;
+	size_t count = 0;
+	for( ; it != end; ++it, ++count ) {
+		if( *it != *(it.operator->()) ) {
+			std::cerr << "Error: dereference operator returns " << *it << ", "
+				<< "but dereferencing the pointer operator returns " << (*(it.operator->()))
+				<< "\n";
+			rc = grb::FAILED;
+			return;
+		}
+		if( *it != retained ) {
+			std::cerr << "Error: found " << *it << ", "
+				<< "but only value(s) " << retained << " were expected\n";
+			rc = grb::FAILED;
+			return;
+		}
+	}
+	if( count != expected ) {
+		std::cerr << "Error: count should be " << n << ", "
+			<< "but the filtered iterator returned " << count << " elements instead\n";
+		rc = grb::FAILED;
+	}
+}
+
+
+void grb_program( const size_t &n, grb::RC &rc ) {
+	size_t expected;
+	grb::RC local_rc;
+	rc = grb::SUCCESS;
+
+	// first fill some vector v with a range of numbers from 0 to n-1 (inclusive)
+	std::vector< size_t > v( n );
+	std::iota( v.begin(), v.end(), 0 );
+
+	// test for filtering out the number 7
+	{
+		auto itPair = grb::utils::iterators::createIteratorFilter(
+			v.begin(), v.end(),
+			[] (const size_t val) { return val == 7; }
+		);
+		if( n < 8 ) {
+			expected = n;
+		} else {
+			expected = n - 1;
+		}
+		testOneOut( n, 7, expected, itPair.first, itPair.second, local_rc );
+		if( local_rc != grb::SUCCESS ) {
+			std::cerr << "Test 1 FAILED\n";
+			rc = grb::FAILED;
+		}
+	}
+
+	// similar test but with one or both iterators copied
+	{
+		auto itPair = grb::utils::iterators::createIteratorFilter(
+			v.begin(), v.end(),
+			[] (const size_t val) { return val == 17; }
+		);
+		auto beginCopy = itPair.first;
+		auto beginCopy2 = itPair.first;
+		auto endCopy = itPair.second;
+		if( n < 18 ) {
+			expected = n;
+		} else {
+			expected = n - 1;
+		}
+		testOneOut( n, 17, expected, beginCopy, itPair.second, local_rc );
+		if( local_rc != grb::SUCCESS ) {
+			std::cerr << "Test 2 FAILED\n";
```

Comment:
```suggestion
			std::cerr << "Test 2 FAILED: rc is " + grb::toString( rc ) + "\n";
```

Everywhere in this file ?

Reply:
I did this everywhere (will commit soon); thanks!